mysql-test/t/mysql_upgrade_fails_toku_80.test

```diff
@@ -0,0 +1,39 @@
+--source include/big_test.inc
+--source include/not_valgrind.inc
+--source include/have_debug.inc
+--source include/have_innodb_16k.inc
+
+# Stop the default mtr server
+
+--let $shutdown_server_timeout = 300
+--source include/shutdown_mysqld.inc
+
+# Setup
+
+--let $MYSQLD_DATADIR1= $MYSQL_TMP_DIR/data_ps8173
+--let $ZIP_BASENAME= data_ps8173.zip
+--let $ZIP_FILE= $MYSQLTEST_VARDIR/std_data/$ZIP_BASENAME
+
+--copy_file $ZIP_FILE $MYSQL_TMP_DIR/$ZIP_BASENAME
+--file_exists $MYSQL_TMP_DIR/$ZIP_BASENAME
+--exec unzip -qo $MYSQL_TMP_DIR/$ZIP_BASENAME -d $MYSQL_TMP_DIR
+--remove_file $MYSQL_TMP_DIR/$ZIP_BASENAME
+
+# Test the upgrade fails when previous is 8.0.x and uses TokuDb
+
+--let $MYSQLD_LOG= $MYSQLTEST_VARDIR/log/mysql_ps8173_upgrade_complete.log
+--replace_result $MYSQLD MYSQLD $MYSQLD_DATADIR1 MYSQLD_DATADIR1 $MYSQLD_LOG MYSQLD_LOG
+--error 1
+--exec $MYSQLD --no-defaults --secure-file-priv="" --datadir=$MYSQLD_DATADIR1 --log-error=$MYSQLD_LOG
```

Comment:
@ldonoso is the datadir from failed upgrade (to 8.0) usable by 5.7 server? (please do manual check)

Reply:
@satya-bodapati , it works correctly. To find out I have:

- Modified the mtr test so we do not delete the data dir after running `mysql_upgrade_fails_toku_80.test`
- Initiated an instance of 5.7 using the data dir generated in the test above
- Connect to the instance and review that the schemas are working